skops/card/_model_card.py

```diff
@@ -7,55 +7,65 @@
 import skops
 
 
-def _extract_estimator_config(model):
-    """Extracts estimator configuration and renders them into a vertical table.
-
-    Parameters
-    ----------
-        model (estimator): scikit-learn pipeline or model.
-
-    Returns
-    -------
-    str:
-        Markdown table of hyperparameters.
-    """
-    hyperparameter_dict = model.get_params(deep=True)
-    table = "| Hyperparameters | Value |\n| :-- | :-- |\n"
-    for hyperparameter, value in hyperparameter_dict.items():
-        table += f"| {hyperparameter} | {value} |\n"
-    return table
-
-
-def create_model_card(
-    model,
-    card_data,
-    **card_kwargs,
-):
-    """Creates a model card for the model and saves it to the target directory.
-
-    Parameters:
-    ----------
-    model: estimator
-        scikit-learn compatible estimator.
-    card_data: CardData
-        CardData object.
-    card_kwargs:
-        Card kwargs are information you can pass to fill in the sections of the
-        card template, e.g. model_description, citation_bibtex, get_started_code.
-    """
-    ROOT = skops.__path__
-    model_plot = re.sub(r"\n\s+", "", str(estimator_html_repr(model)))
-    hyperparameter_table = _extract_estimator_config(model)
-    card_data.library_name = "sklearn"
-    template_path = card_kwargs.get("template_path")
-    if template_path is None:
-        template_path = os.path.join(f"{ROOT[0]}", "card", "default_template.md")
-    card_kwargs["template_path"] = template_path
-    card = ModelCard.from_template(
-        card_data=card_data,
-        hyperparameter_table=hyperparameter_table,
-        model_plot=model_plot,
-        **card_kwargs,
-    )
-
-    return card
+class Card:
+    def __init__(self, model):
+        self.model = model
+        self.hyperparameters_table = self.extract_estimator_config(self.model)
+        self.model_plot = re.sub(r"\n\s+", "", str(estimator_html_repr(model)))
+        self.template_sections = {}
+
+    def add(self, section, value):
+        """Takes values to fill model card template.
+        Parameters:
+        ----------
+        section: str
+                Section in the template.
+        value: str
+                Value to fill in the section.
+        """
+        self.template_sections[section] = value
+        return self
```

Comment:
I like the API, but I would suggest changing this to:

```suggestion
    def add(self, **kwargs):
        """Takes values to fill model card template.
        Parameters:
        ----------
        **kwargs : dict
            Parameters to be set for the model card. These parameters
            need to be sections of the underlying `jinja` template used.

        """
        for section, value in kwargs.items():
            self.template_sections[section] = value
        return self
```

This would allow users to set multiple values on each call, yet allowing them to also chain the `add` calls as they wish.